frontend/src/app/pages/exercises/exercise/shared/trainer-map-editor/trainer-map-editor.component.ts

```diff
@@ -82,4 +90,28 @@ export class TrainerMapEditorComponent {
             template: vehicleTemplate,
         });
     }
+
+    public importingTemplates = false;
+    public async importPartialExport(fileList: FileList) {
+        try {
+            this.importingTemplates = true;
+            const importedText = await fileList.item(0)?.text();
+            if (importedText === undefined) {
+                // The file dialog has been aborted.
+                return;
+            }
+            const importedPlainObject = JSON.parse(importedText) as object;
+            const importedInstance = plainToInstance(
+                PartialExport,
+                importedPlainObject
+            );
+            const migratedInstance = migratePartialExport(importedInstance);
```

Comment:
```suggestion
            const importedPlainObject = JSON.parse(importedText) as object;
            const migratedPartialExport = migratePartialExport(importedPlainObject);
            if (!isValid(plainToInstance(
                PartialExport,
                migratedPartialExport 
            ))) {
                 throw Error('PartialExport is invalid');
            }
```

Reply:
At the moment, you are passing an instance to the modal (instead of a plain object) and, therefore, to the action. I would suspect this to crash, if this instance is later modified in a reducer (e.g., a template gets renamed) as `immer.js` would find a prototype in the template object.

Remember that all the `.create()-factory-functions were there to *not* have instances in the state.

FYI: Instead of the factory functions, one could also add a token to those classes in which immer should ignore the prototypes. But having plain objects without prototypes could be more performant (https://github.com/immerjs/immer/pull/941) and maybe less error-prone. #531